.devcontainer/post-create.sh

```diff
@@ -6,3 +6,9 @@ cargo install cargo-insta
 cargo fetch
 
 pip install maturin pre-commit
+
+# configure git to be able to run from the devcontainer
+git config --global --add safe.directory /workspaces/ruff
+
+# ensure pre-commit is configured to run
+pre-commit install
```

Comment:
I'm a bit hesitant to enforce pre-commit to contributors inside the dev containers or more generally I'm hesitant to install anything other than the required tools to be able to contribute to Ruff. I, personally, don't install the hooks and run them manually as I might commit a lot.

Reply:
i guess it ultimately is comes down to the prevailing preference.  I generally prefer about anything I might want to be installed, and I can always disable.  BUT, I understand the alternative, so can close this PR.

Reply:
Thank you for understanding :)